linux/src/async_socket_linux.c

```diff
@@ -108,8 +108,21 @@ static int on_socket_send(void* context, ASYNC_SOCKET_HANDLE async_socket, const
     }
     else
     {
-        // Codes_SRS_ASYNC_SOCKET_LINUX_11_052: [ on_socket_send shall attempt to send the data by calling send with the MSG_NOSIGNAL flag to ensure SIGPIPE is not generated on errors. ]
-        result = send(async_socket->socket_handle, buf, len, MSG_NOSIGNAL);
+        // Codes_SRS_ASYNC_SOCKET_LINUX_11_052: [ on_socket_send shall attempt to send the data by calling socket_transport_send with the MSG_NOSIGNAL flag to ensure SIGPIPE is not generated on errors. ]
+        SOCKET_BUFFER input_buf;
+        input_buf.buffer = buf;
+        input_buf.length = len;
+        uint32_t bytes_sent;
+        if(socket_transport_send(async_socket->socket_transport_handle, &input_buf, 1, &bytes_sent, MSG_NOSIGNAL, NULL) != SOCKET_SEND_OK)
+        {
+            result = -1;
+            LogError("socket_transport_send failed.");
```

Comment:
>." [](http://example.com/codeflow?start=50&length=2)

In this LogError you want to put some information that will give us clues on what failed, so you want to make the LogError more like this:

LogError("socket_transport_send failed input_buf.buffer: %p, input_buf.length: %" Priu32 ".", input_buf.buffer, input_buf.length);